lib/filterx/filterx-object.c

```diff
@@ -68,6 +93,9 @@ filterx_object_unref(FilterXObject *self)
   if (!self)
     return;
 
+  if (self->ref_cnt == FILTERX_OBJECT_MAGIC_BIAS)
```

Comment:
As a follow-up to the previous comment: non-atomic ref counts caused a lot of trouble considering long-term maintenance once we forgot about them and started using objects across thread boundaries.

Why don't we use an atomic ref-count instead of building a freezing mechanism on top of non-atomic counters?

Reply:
Let's discuss this IRL tomorrow.

Reply:
We concluded that we would add an assert on cross-thread use of FilterXObject